x/ccv/provider/keeper/partial_set_security.go

```diff
@@ -308,11 +308,54 @@ func (k Keeper) CanValidateChain(ctx sdk.Context, chainID string, providerAddr t
 			!k.IsDenylisted(ctx, chainID, providerAddr))
 }
 
+// FulfillsMinStake returns true if the validator `providerAddr` has enough stake to validate chain `chainID`
+// by checking its staked tokens against the minimum stake required to validate the chain.
+func (k Keeper) FulfillsMinStake(ctx sdk.Context, chainID string, providerAddr types.ProviderConsAddress) bool {
+	minStake, found := k.GetMinStake(ctx, chainID)
+	if !found {
+		return true
+	}
+
+	validator, err := k.stakingKeeper.GetValidatorByConsAddr(ctx, providerAddr.Address)
+	if err != nil {
+		return false
+	}
+
+	// validator has enough stake to validate the chain
+	return validator.GetTokens().GTE(math.NewInt(minStake))
+}
+
 // ComputeNextValidators computes the validators for the upcoming epoch based on the currently `bondedValidators`.
 func (k Keeper) ComputeNextValidators(ctx sdk.Context, chainID string, bondedValidators []stakingtypes.Validator) []types.ConsensusValidator {
+	// sort the bonded validators by power in descending order
+	sort.Slice(bondedValidators, func(i, j int) bool {
+		iTokens := bondedValidators[i].GetTokens()
+		jTokens := bondedValidators[j].GetTokens()
+		result := iTokens.GT(jTokens)
+		return result
```

Comment:
Why not simply do this in just one line: `return bondedValidators[i].GetTokens().GT(bondedValidators[j].GetTokens())`?

Could `sort.Slice` lead to different orderings if 2 validators have the same number of tokens? For example, if we have validator `A (tokens = 50)` and `B  (tokens = 50)` that we end up with in one run with `A, B` and in another with `B, A`? I do not think this is the case (we already use `sort.Slice` extensively but might be good to verify).

Reply:
Woops, this was for debugging. Thanks for the catch!